### DISCUSSION TITLE
Is it possible to define default role for new OIDC users?

### DISCUSSION THREAD

**Discussion by sseide** [photoprism/photoprism#4438] [Q&A] [answered]
Connecting Photoprism to an OpenID connect provider and allowing registration of new OIDC authenticated users (`PHOTOPRISM_OIDC_REGISTER: "true"`) automatically create these users on first login. These users have the role "Guest" assigned.

Is there any configuration option to define the role these users should get as default (instead of Guest)? I searched the docs but did not find anything.

Or do i have to modify the users manually after first login with `photoprism user mod --role Users <username>`?

Thanks in advance

(PhotoPrism® Essentials Build 240711-2197af848)

**Comment by lastzero** [comment, marked as answer]
For security reasons, our *Personal Editions* currently default to the Guest role, which administrators can then upgrade after checking the eligibility of newly registered accounts. If you run our *Pro Edition* in a trusted corporate network with appropriate security measures - including for the OIDC provider - it can be configured to give new accounts a higher authorization level by default.

Please note in this context that using an external Identity Provider for authorization, and not just for authentication, can easily lead to security issues such as the following, for which we do not want to get a CVE assigned nor do we want to be responsible for any private pictures of our users getting leaked as a result:

- https://www.microsoft.com/en-us/security/blog/2024/07/29/ransomware-operators-exploit-esxi-hypervisor-vulnerability-for-mass-encryption/
- https://support.broadcom.com/web/ecx/support-content-notification/-/external/content/SecurityAdvisories/0/24505

**Reply by lastzero** [reply]
As this question might come up again in the future, I have added the answer to our FAQs:

- https://docs.photoprism.app/getting-started/advanced/openid-connect/#is-it-possible-to-set-a-default-role-for-new-oidc-users
